compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -1035,6 +1035,15 @@ pub enum Rvalue<'tcx> {
     /// initialized but its content as uninitialized. Like other pointer casts, this in general
     /// affects alias analysis.
     ShallowInitBox(Operand<'tcx>, Ty<'tcx>),
+
+    /// A CopyForDeref is equivalent to a read from a place at the
+    /// codegen level, but is treated specially by drop elaboration. When such a read happens, it
+    /// is guaranteed (via nature of deref_seperator) that the only use of the returned value is a deref operation, immediately
+    /// followed by one or more projections. Drop elaboration treats this rvalue as if the
```

Comment:
Why are the projections relevant? What is the problem if the only use of this value is in `*v` without a projection?

Reply:
Without projection, we wouldn't even run `deref_separator` on it so it wouldn't be used.

Reply:
Sure but there's nothing *wrong* with that situation, right? If it somehow happened, all passes could handle it fine?
I just want to figure out if there is a correctness argument here that relies on "at least one projection", which would seem odd.

Reply:
I never considered the _if_ part so I made some local changes that makes sure it will work no matter what , I will push it alongside CI fix (Once I solve it 😭 ).